server/app/services/applications/PdfExporterService.java

```diff
@@ -0,0 +1,51 @@
+package services.applications;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.inject.Inject;
+import com.itextpdf.text.DocumentException;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import models.Application;
+import services.DeploymentType;
+import services.export.PdfExporter;
+import services.program.ProgramDefinition;
+
+/** The service responsible for exporting a PDF file */
+public final class PdfExporterService {
+
+  private final ApplicationService applicationService;
+  private final PdfExporter pdfExporter;
+
+  @Inject
+  PdfExporterService(
+      ApplicationService applicationService,
+      Config configuration,
+      DeploymentType deploymentType,
+      PdfExporter pdfExporter) {
+    this.applicationService = checkNotNull(applicationService);
+    this.pdfExporter = checkNotNull(pdfExporter);
+
+    checkNotNull(configuration);
+    checkNotNull(deploymentType);
+  }
+
+  public PdfExporter.InMemoryPdf generatePdf(Long applicationId, ProgramDefinition program) {
```

Comment:
**suggestion (blocking):** pass the application object in instead of the ID, this class should not be responsible for retrieving data from the DB